doc/_extensions/zephyr/api_overview.py

```diff
@@ -21,156 +43,121 @@ class ApiOverview(SphinxDirective):
 
     Configuration options:
 
-    api_overview_doxygen_xml_dir: Doxygen xml output directory
-    api_overview_doxygen_base_url: Doxygen base html directory
+    api_overview_doxygen_out_dir: Doxygen output directory
     """
 
     def run(self):
-        return [self.env.api_overview_table]
-
+        groups = parse_xml_dir(self.config.api_overview_doxygen_out_dir + "/xml")
 
-def get_group(innergroup, all_groups):
-    try:
-        return [
+        toplevel = [
             g
-            for g in all_groups
-            if g.get_compounddef()[0].get_id() == innergroup.get_refid()
-        ][0]
-    except IndexError as e:
-        raise Exception(f"Unexpected group {innergroup.get_refid()}") from e
+            for g in groups
+            if g.get_compounddef()[0].get_id()
+            not in [
+                i.get_refid()
+                for h in [j.get_compounddef()[0].get_innergroup() for j in groups]
+                for i in h
+            ]
```

Comment:
It's quite a mouthful. Could use a comment, or slightly more explicit variable names?